src/Google.Storage.V1/StorageClientImpl.CreateBucket.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Storage.v1;
+using Google.Apis.Storage.v1.Data;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Storage.V1
+{
+    public partial class StorageClientImpl
+    {
+        /// <inheritdoc />
+        public override Bucket CreateBucket(string projectId, string bucket, CreateBucketOptions options = null) =>
+            CreateCreateBucketRequest(projectId, bucket, options).Execute();
+
+        /// <inheritdoc />
+        public override Task<Bucket> CreateBucketAsync(string projectId, string bucket, CreateBucketOptions options = null,
+            CancellationToken cancellationToken = default(CancellationToken)) =>
+            CreateCreateBucketRequest(projectId, bucket, options).ExecuteAsync(cancellationToken);
+
+        /// <inheritdoc />
+        public override Bucket CreateBucket(string projectId, Bucket bucket, CreateBucketOptions options = null) =>
+            CreateCreateBucketRequest(projectId, bucket, options).Execute();
+
+        /// <inheritdoc />
+        public override Task<Bucket> CreateBucketAsync(string projectId, Bucket bucket, CreateBucketOptions options = null, CancellationToken cancellationToken = default(CancellationToken)) =>
+            CreateCreateBucketRequest(projectId, bucket, options).ExecuteAsync(cancellationToken);
+
+        private BucketsResource.InsertRequest CreateCreateBucketRequest(string projectId, string bucket, CreateBucketOptions options)
```

Comment:
There are slightly different bits of validation going on. If we specify a null bucket name, we want an ArgumentNullException not an ArgumentException.